src/reistijden_v1/views.py

```diff
@@ -0,0 +1,104 @@
+        'latitude': src_d['lane']['camera']['coordinates']['@latitude'],
+        'longitude': src_d['lane']['camera']['coordinates']['@longitude'],
+        'lane_number': src_d['lane']['camera']['lane_number'],
+        'status': src_d['lane']['camera']['status'],
+        'view_direction': src_d['lane']['camera']['view_direction'],
+    }
+
+
+def travel_time_src_to_dict(src_d):
+    return {
+        'travel_time_type': src_d['@travel_time_type'],
+        'estimation_type': src_d['@estimation_type'],
+        'travel_time': src_d['travel_time'],
+        'traffic_speed': src_d['traffic_speed'],
+    }
+
+
+def traffic_flow_src_to_dict(src_d):
+    measured_flow = src_d['measured_flow']
+    return {
+        'specific_lane': measured_flow['@specific_lane'],
+        'vehicle_flow': measured_flow['vehicle_flow'],
+        'category_count': measured_flow['number_of_input_values_used']['category']['@count'],
+        'category_type': measured_flow['number_of_input_values_used']['category']['@type'],
+    }
+
+
+def get_location_from_site_ref(site_ref):
+    if 'location' in site_ref:
+        return [location_src_to_dict(site_ref['location'])]
+    else:
+        return [location_src_to_dict(d) for d in site_ref['location_contained_in_itinerary']['location']]
+
+
+def measurement_src_to_dict(src_d):
+    site_ref = src_d['measurement_site_reference']
+    return {
+        'measurement_site_reference_id': site_ref['@id'],
+        'measurement_site_reference_version': site_ref['@version'],
+        'measurement_site_name': site_ref.get('measurement_site_name'),
+        'measurement_site_type': site_ref['measurement_site_type'],
+        'length': site_ref.get('length'),
+        'locations': get_location_from_site_ref(site_ref),
+        'travel_times': [travel_time_src_to_dict(src_d['travel_time_data'])] if 'travel_time_data' in src_d else [],
+        'individual_travel_times': [d for d in src_d['individual_travel_time_data']] if 'individual_travel_time_data' in src_d else [],
+        'traffic_flows': [traffic_flow_src_to_dict(src_d['traffic_flow_data'])] if 'traffic_flow_data' in src_d else [],
+    }
+
+
+def restructure_data(xml_str):
+    data_dict = humps.decamelize(xmltodict.parse(xml_str.strip()))
+    publication_src = data_dict['amsterdam_travel_times']['payload_publication']
+    return {
+        'publication_type': publication_src['@type'],
+        'publication_reference_id': publication_src['publication_reference']['@id'],
+        'publication_reference_version': publication_src['publication_reference']['@version'],
+        'publication_time': publication_src['publication_time'],
+        'measurement_start_time': publication_src['measurement_period']['measurement_start_time'],
+        'measurement_end_time': publication_src['measurement_period']['measurement_end_time'],
+        'measurements': [measurement_src_to_dict(d) for d in publication_src['site_measurements']],
+    }
```

Comment:
Dit is eigenlijk de verantwoordelijkheid van de serializer, maar gezien de input die je krijgt snap ik deze oplossing wel. Het is erg clean en heel duidelijk :+1:

Reply:
Een oplossing zou kunnen zijn om een extra serializer te maken die exact bovenstaande doet, maar ik betwijfel of die code net zo simpel zou worden als bovenstaande.

Reply:
Je hebt helemaal gelijk. Maar gezien ik op deze manier de validation van de standaard serializer kan gebruiken en ik eventuele structuur-errors hier afvang met een try-catch vind ik (zoals je zelf al aangeeft) dit eigenlijk een schonere oplossing.